generators/service-watson-discovery/templates/python/instrumentation.py

```diff
@@ -1,10 +1,25 @@
 from ibmcloudenv import IBMCloudEnv
 from watson_developer_cloud import DiscoveryV1
 
-discovery = DiscoveryV1(
-    username=IBMCloudEnv.getString('watson_discovery_username'),
-    password=IBMCloudEnv.getString('watson_discovery_password'),
-    version='2016-12-01')
+if IBMCloudEnv.getString('watson_discovery_apikey'):
+    iam_url = 'https://iam.stage1.bluemix.net/identity/token' if 'gateway-s.' in params.url else 'https://iam.bluemix.net/identity/token'
+    iam_apikey = api_key=IBMCloudEnv.getString('watson_discovery_apikey')
```

Comment:
instead of using a variable, we can use `IBMCloudEnv.getString('watson_discovery_apikey')` in call for service:
```
DiscoveryV1(
       url=IBMCloudEnv.getString('watson_discovery_url'),
       iam_api_key=IBMCloudEnv.getString('watson_discovery_apikey'),
       iam_url=iam_url,
       version='2018-03-05')
```